batch/batch/server/server.py

```diff
@@ -259,6 +259,11 @@ def _has_next_task(self):
         uri = await write_gs_log_file(app['blocking_pool'], INSTANCE_ID, self.id, task_name, log)
         await db.jobs.update_log_uri(self.id, task_name, uri)
 
+    async def _delete_logs(self):
+        for idx, jt in enumerate(self._tasks):
+            if idx < self._task_idx:
+                await delete_gs_log_file(app['blocking_pool'], INSTANCE_ID, self.id, jt.name)
```

Comment:
I'm worried about what happens when this fails.  Say we've deleted several of the logs, and a later one times out.  That will throw out of the cleanup loop, and we'll try again later.  When we go to delete the first logs again, we'll get a 404 ... throwing out of the cleanup loop.

In other words, I think this should ignore 404.  Then I think I'm happy.